### DISCUSSION TITLE
making request GRAPH QL api

### DISCUSSION THREAD

**Discussion by sunilkumarnagoore** [httprb/http#724] [General]
how to make use this http gem to make request for GRAPH QL API

**Comment by skatkov** [comment]
A great question to ask in discussions.

**Comment by ixti** [comment]
@sunilkumarnagoore GraphQL is a protocol on top of HTTP. This gem provides HTTP client. You can find details on how to make GraphQL queries here: https://graphql.org/learn/serving-over-http/#http-methods-headers-and-body

Here's a small example:

``` ruby
require "bundler/inline"

gemfile do
  source "https://rubygems.org"

  gem "http"
end

authorization = "bearer #{ENV.fetch('GITHUB_ACCESS_TOKEN')}"
query = <<~GRAPHQL
  query {
    repository(owner: "httprb", name: "http") {
      description
    }
  }
GRAPHQL

puts HTTP \
  .headers("Authorization" => authorization) \
  .post("https://api.github.com/graphql", json: { query: query }) \
  .parse(:json)
```
